src/command_modules/azure-cli-cognitiveservices/azure/cli/command_modules/cognitiveservices/_params.py

```diff
@@ -69,13 +74,64 @@ def _is_suspected_json(string):
 )
 
 
+def _sku_filter(cmd, namespace):
+    """
+    Get a list of ResourceSku and filter by existing conditions: 'kind', 'location' and 'sku_name'
+    """
+    kind = getattr(namespace, 'kind', None)
+    location = getattr(namespace, 'location', None)
+    sku_name = getattr(namespace, 'sku_name', None)
+
+    def _filter_sku(_sku):
+        if sku_name is not None:
+            if _sku.name != sku_name:
+                return False
+        if kind is not None:
+            if _sku.kind != kind:
+                return False
+        if location is not None:
+            if location.lower() not in [x.lower() for x in _sku.locations]:
+                return False
+        return True
+
+    return [x for x in cf_resource_skus(cmd.cli_ctx).list() if _filter_sku(x)]
+
+
+@Completer
+def sku_name_completer(cmd, prefix, namespace, **kwargs):  # pylint: disable=unused-argument
+    names = set([x.name for x in _sku_filter(cmd, namespace)])
+    return sorted(list(names))
+
+
+@Completer
+def kind_completer(cmd, prefix, namespace, **kwargs):  # pylint: disable=unused-argument
+    kinds = set([x.kind for x in _sku_filter(cmd, namespace)])
+    return sorted(list(kinds))
+
+
+@Completer
+def location_completer(cmd, prefix, namespace, **kwargs):  # pylint: disable=unused-argument
+    names = set(sum([x.locations for x in _sku_filter(cmd, namespace)], []))
+    return [x.lower() for x in sorted(list(names))]
+
+
+def get_cognitiveservices_location_type(ctx):
+    _location_type = get_location_type(ctx)
+    _location_type.update(completer=location_completer)
+    return _location_type
+
+
 def load_arguments(self, _):
     with self.argument_context('cognitiveservices') as c:
         c.argument('account_name', arg_type=name_arg_type, help='cognitive service account name',
                    completer=get_resource_name_completion_list('Microsoft.CognitiveServices/accounts'))
-        c.argument('resource_group_name', arg_type=resource_group_name_type)
-        c.argument('sku_name', options_list=['--sku'], help='the Sku of cognitive services account')
-        c.argument('kind', help='the API name of cognitive services account')
+        c.argument('location', arg_type=get_cognitiveservices_location_type(self.cli_ctx))
+        c.argument('resource_group_name', arg_type=resource_group_name_type,
+                   completer=get_resource_group_completion_list)
```

Comment:
`resource_group_name_type` already contains the completer: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/commands/parameters.py#L236

Reply:
okay, will remove this.